tests/ui/macros/attr-empty-expr.stderr

```diff
@@ -1,20 +0,0 @@
-error: removing an expression is not supported in this position
-  --> $DIR/attr-empty-expr.rs:8:13
-   |
-LL |     let _ = #[test] 0;
-   |             ^^^^^^^
-
-error: removing an expression is not supported in this position
```

Comment:
this test was testing for "empty expressions". this is super niche, can only happen with ast-based attributes (only like 10 in the compiler) and none of the examples in this test should actually remove the expression, they should instead report that the target is wrong. I can't find any other examples of node-removing attributes so I think this test is simply superfluous now.